app/webpack/src/table-select-all.js

```diff
@@ -1,54 +1,65 @@
-$(document).ready(function() {
-  if ($('.select-clear-all').length) {
+document.addEventListener('DOMContentLoaded', event => {
+  if (document.querySelector('.select-clear-all')) {
     function resetSelectClearLink(table) {
-      const link = table.find('.select-clear-all');
-      link.attr('tabindex', 0);
-      link.removeClass('select-all clear-all');
+      const link = table.querySelector('.select-clear-all');
+      link.setAttribute('tabindex', 0);
+      link.classList.remove('select-all', 'clear-all');
 
-      if (table.find('input:checked').length) {
-        link.addClass('clear-all').text(link.data('copy-clear'));
+      if (table.querySelector('input:checked')) {
+        link.classList.add('clear-all')
+        link.textContent = link.getAttribute('data-copy-clear');
       }
       else {
-        link.addClass('select-all').text(link.data('copy-select'));
+        link.classList.add('select-all')
+        link.textContent = link.getAttribute('data-copy-select');
       }
     }
 
     function setCategory(row) {
-      let vacantItem = row.find(".table-category-vacant");
-      if (row.find('input:checked').length) {
-        vacantItem.addClass("table-category-preview");
+      let vacantItem = row.querySelector(".table-category-vacant");
+      if (row.querySelector('input:checked')) {
+        vacantItem.classList.add("table-category-preview");
       } else {
-        vacantItem.removeClass("table-category-preview");
+        vacantItem.classList.remove("table-category-preview");
       }
     }
 
-    $('.select-clear-all').click(function() {
-      const table = $(this).parents('table');
-      if($(this).hasClass('clear-all')) {
-        $("#screen-reader-messages").text("All transactions deselected.");  //this adds a message to the message div which is read out by the screen reader
-        table.find('input:checked').prop('checked', false);
-      } else if ($(this).hasClass('select-all')) {
-        table.find('input:not(:checked)').prop('checked', true);
-        $("#screen-reader-messages").text("All transactions selected."); 
+    document.querySelector('.select-clear-all').addEventListener("click", function() {
+      const table = document.querySelector('table');
```

Comment:
I'm guessing there is no risk of grabbing too many table elements as you are now not selecting parents of this ele right?

Reply:
yeah, this is only used in one place and there is only one table on the page, however have now changed it to get the table via the parents once, to avoid making this assumption